yarn-project/pxe/src/database/kv_pxe_database.ts

```diff
@@ -165,27 +165,47 @@ export class KVPxeDatabase implements PxeDatabase {
     }
   }
 
+  *#getAllNonNullifiedNoteIndices(): IterableIterator<number> {
+    for (const [index, _] of this.#notes.entries()) {
+      if (this.#nullifiedNotes.has(index)) {
+        continue;
+      }
+
+      yield index;
+    }
+  }
+
   async getNotes(filter: NoteFilter): Promise<NoteDao[]> {
     const publicKey: PublicKey | undefined = filter.owner
       ? (await this.getCompleteAddress(filter.owner))?.publicKey
       : undefined;
 
-    const initialNoteIds = publicKey
-      ? this.#notesByOwner.getValues(publicKey.toString())
-      : filter.txHash
-      ? this.#notesByTxHash.getValues(filter.txHash.toString())
-      : filter.contractAddress
-      ? this.#notesByContract.getValues(filter.contractAddress.toString())
-      : filter.storageSlot
-      ? this.#notesByStorageSlot.getValues(filter.storageSlot.toString())
-      : undefined;
+    // TODO: refactor how we create the array of candidate note ids once the kv_pxe_database refactor from #3927 is
```

Comment:
is this meant to be resolved in this PR or another. If latter - create issue?

Reply:
It was mostly a remainder that this will happen. I just removed the todo since the refactor is inevitable.